### ISSUE TITLE
bug(material/list): selected state is visually communicated only with color (WCAG 1.4.1)

### ISSUE THREAD

**Issue by zarend** [angular/components#25900]
### Description

single-select list visually communicate selected state with color alone. It should be visually communicated with more than just color.

https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

### Reproduction

1. open https://material2-dev.web.app/list
2. Navigate to "Single Selection list" demo
3. Under "Favorite Grocery", click "Oranges"

### Expected Behavior

selected state of "Oranges" is visually communicated with more than color alone

### Actual Behavior

color of "Oranges" changes to indicate selected state
<img width="351" alt="image" src="https://user-images.githubusercontent.com/7720245/199113523-cb7df5f6-f094-44ac-988c-c26d89c18a02.png">
